lib/player.js

```diff
@@ -2051,14 +2051,17 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     if (!activeVariantTrack) {
       initialVariant = this.chooseVariant_();
       goog.asserts.assert(initialVariant, 'Must choose an initial variant!');
+    }
 
-      // Lazy-load it, so we will have enough info to make the playhead.
-      const createSegmentIndexPromises = [];
-      for (const stream of [initialVariant.video, initialVariant.audio]) {
-        if (stream) {
-          createSegmentIndexPromises.push(stream.createSegmentIndex());
-        }
+    // Lazy-load the stream, so we will have enough info to make the playhead.
+    const createSegmentIndexPromises = [];
+    const toLazyLoad = activeVariantTrack || initialVariant;
+    for (const stream of [toLazyLoad.video, toLazyLoad.audio]) {
+      if (stream && !stream.segmentIndex) {
+        createSegmentIndexPromises.push(stream.createSegmentIndex());
       }
+    }
+    if (createSegmentIndexPromises.length > 0) {
```

Comment:
This is fine, but JFYI, `await Promise.all([])` (empty array) is equivalent to `await Promise.resolve()`.  It will schedule a microtask and return control back to the interpreter loop, but it won't hurt anything.

Reply:
I was worried that it might change the results of some test to have another await in the middle, so I put that in preemptively.
It's almost definitely unnecessary, I agree.